File: 1520-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/number-of-steps-to-reduce-a-number-in-binary-representation-to-one.c
#include <stdio.h>
#include <string.h>

int numSteps(char* s) {
    int l = strlen(s) - 1;
    int carry = 0;
    int count = 0;
    
    while (l > 0) {
        if (s[l] - '0' + carry == 0) {
            carry = 0;
            count++;
        } else if (s[l] - '0' + carry == 2) {
            carry = 1;
            count++;
        } else {
            carry = 1;
            count += 2;
        }
        l--;
    }
    
    if (carry == 1) {
        count++;
    }
    
    return count;
}

File: 1520-number-of-steps-to-reduce-a-number-in-binary-representation-to-one/number-of-steps-to-reduce-a-number-in-binary-representation-to-one.py
class Solution(object):
    def numSteps(self, s):
        s = int(s, 2)
        cnt = 0

        while s != 1:
            if s % 2 == 0:
                s //= 2
            else:
                s += 1
            cnt += 1
        
        return cnt
